File: 12challenge/main.py
class Product:
    def __init__(self, price:float):
        self.price = price
        self.discount = 0.05
        self.final_price = self.price - (self.price * self.discount)
    
    def __str__(self):
        return f'Product price: {self.price:.2f} €\nPromotional price: {self.final_price:.2f} €'
    

product = Product(price = float(input("Enter the price of the product: ")))
print(product)

File: 12challenge/main.go
package main

import "fmt"

type Product struct {
    price float64
    discount float64
    finalPrice float64
}

func (p *Product) calculateFinalPrice(){
    p.finalPrice = p.price - (p.price * p.discount)
}

func (p Product) printProduct(){
    fmt.Printf("Product price: %.2f\nPromotional price: %.2f\n",p.price,p.finalPrice)
}

func main() {
    var price float64
    fmt.Print("Enter the price of the product: ")
    fmt.Scan(&price)

    product := &Product{price:price,discount:0.05}
    product.calculateFinalPrice()
    product.printProduct()
}
